Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealityProfileUtility.cs

```diff
@@ -0,0 +1,157 @@
+        private static Dictionary<Type, GUIContent[]> profileContentCaches = new Dictionary<Type, GUIContent[]>();
+        private static Dictionary<Type, Type[]> profileTypesForServiceCaches = new Dictionary<Type, Type[]>();
+
+        /// <summary>
+        /// Returns an array of profiles that match profile type.
+        /// </summary>
+        /// <param name="profileType"></param>
+        /// <returns></returns>
+        public static ScriptableObject[] GetProfilesOfType(Type profileType)
+        {
+            ScriptableObject[] profilesOfType = null;
+            if (!profileCaches.TryGetValue(profileType, out profilesOfType))
+            {
+                profilesOfType = ScriptableObjectExtensions.GetAllInstances(profileType);
+                profileCaches.Add(profileType, profilesOfType);
+            }
+            return profilesOfType;
+        }
+
+        /// <summary>
+        /// Returns an array of GUIContent for use in a dropdown for a type of profile.
+        /// Includes a (None) option at the start. This means that the array length will always be 1 greater than the available profiles.
+        /// </summary>
+        /// <param name="profileType"></param>
+        /// <returns></returns>
+        public static GUIContent[] GetProfilePopupOptionsByType(Type profileType)
+        {
+            GUIContent[] profileContent = null;
+            if (!profileContentCaches.TryGetValue(profileType, out profileContent))
+            {
+                ScriptableObject[] profilesOfType = GetProfilesOfType(profileType);
+                profileContent = new GUIContent[profilesOfType.Length + 1];
+                profileContent[0] = new GUIContent("(None)");
+                for (int i = 0; i < profilesOfType.Length; i++)
+                {
+                    profileContent[i + 1] = new GUIContent(profilesOfType[i].name);
+                }
+                profileContentCaches.Add(profileType, profileContent);
+            }
+            return profileContent;
+        }
+
+        /// <summary>
+        /// Returns true if the given profile type is designed to configure the given service.
+        /// </summary>
+        public static bool IsProfileForService(Type profileType, Type serviceType)
+        {
+            foreach (MixedRealityServiceProfileAttribute serviceProfileAttribute in profileType.GetCustomAttributes(typeof(MixedRealityServiceProfileAttribute), true))
+            {
+                if (serviceProfileAttribute.ServiceType.IsAssignableFrom(serviceType))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Returns true if profile is NOT a BaseMixedRealityProfile class type.
+        /// </summary>
+        /// <param name="profileTypeName"></param>
+        /// <returns></returns>
+        public static bool IsConcreteProfileType(String profileTypeName)
```

Comment:
maybe this should use the Type class instead of doing strings, seems like could easily break..especially if class names change etc

Reply:
Probably a good idea, but my goal was to mess with existing functionality as little as possible. I'd want to handle this in a separate PR that was less performance-focused.

Reply:
fair enough, but can we at least change it to use types while here. It doesn't need the additional if type.isclass or type.isabstract etc
public static bool IsConcreteProfileType(Type profileType)
{
return profileType != typeof(BaseMRClass);
}